File: caddisfly-app/app/src/mainExt/java/org/akvo/caddisfly/common/AppConstants.java
/*
 * Copyright (C) Stichting Akvo (Akvo Foundation)
 *
 * This file is part of Akvo Caddisfly.
 *
 * Akvo Caddisfly is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Akvo Caddisfly is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Akvo Caddisfly. If not, see <http://www.gnu.org/licenses/>.
 */

package org.akvo.caddisfly.common;

/**
 * Global Configuration settings for the app.
 */
public final class AppConstants {

    /**
     * Date on which the app version will expire.
     * This is to ensure that installs from apk meant for testing only cannot be used for too long.
     */
    public static final int APP_EXPIRY_DAY = 15;
    public static final int APP_EXPIRY_MONTH = 12;
    public static final int APP_EXPIRY_YEAR = 2017;

    public static final String APP_FOLDER = "Akvo Caddisfly";
    public static final String FFEM_FOLDER = "";

    private AppConstants() {
    }

}

File: caddisfly-app/app/src/main/java/org/akvo/caddisfly/common/AppConstants.kt
/*
 * Copyright (C) Stichting Akvo (Akvo Foundation)
 *
 * This file is part of Akvo Caddisfly.
 *
 * Akvo Caddisfly is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Akvo Caddisfly is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Akvo Caddisfly. If not, see <http://www.gnu.org/licenses/>.
 */
package org.akvo.caddisfly.common

import org.akvo.caddisfly.BuildConfig

/**
 * Global Configuration settings for the app.
 */
object AppConstants {
    /**
     * Url to policies and terms
     */
    const val TERMS_OF_USE_URL = "https://akvo.org/help/akvo-policies-and-terms-2/"

    /**
     * The intent action string used by the caddisfly question type.
     */
    const val EXTERNAL_APP_ACTION = "org.akvo.flow.action.caddisfly"
    /**
     * Uri for photos from built in camera.
     */
    const val FILE_PROVIDER_AUTHORITY_URI = BuildConfig.APPLICATION_ID + ".fileprovider"
    /**
     * To launch Flow app.
     */
    const val FLOW_SURVEY_PACKAGE_NAME = "org.akvo.flow"
}